soroban-env-host/src/test/crypto.rs

```diff
@@ -255,3 +254,177 @@ fn recover_ecdsa_secp256k1_key_test() {
     )
         .err().unwrap()));
 }
+
+#[cfg(feature = "next")]
+#[test]
+fn test_secp256r1_signature_verification() -> Result<(), HostError> {
```

Comment:
Can we also add a few scenarios with large input buffers?

Reply:
The inputs must be `ByteObject` that contains bytes of exact size: 65 bytes `public_key`, 32 bytes `msg_digest` and 64 bytes `signature`. If the length is not correct it gets rejected right away. There is already a test for every incorrect input length in the tests I added.